pkg/virt-launcher/virtwrap/manager_test.go

```diff
@@ -1622,6 +1624,95 @@ var _ = Describe("Manager", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(domSpec).ToNot(BeNil())
 		})
+
+		Context("on call to GetGuestOSInfo", func() {
+			var libvirtmanager DomainManager
+			var agentStore agentpoller.AsyncAgentStore
+
+			BeforeEach(func() {
+				agentStore = agentpoller.NewAsyncAgentStore()
+				libvirtmanager, _ = NewLibvirtDomainManager(mockConn, testVirtShareDir, nil, 0, &agentStore, "/usr/share/OVMF")
+			})
+
+			It("should return nil when no osInfo exists in the cache", func() {
+				osInfo := libvirtmanager.GetGuestOSInfo()
+				Expect(osInfo).To(BeNil())
+			})
+
+			It("should return osInfo when osInfo exists in the cache", func() {
+				fakeInfo := api.GuestOSInfo{
+					Name: "TestGuestOSName",
+				}
+
+				agentStore.Store(agentpoller.GET_OSINFO, fakeInfo)
```

Comment:
This is part of the test preparation, so please place it in the same block as the `fakeInfo` (i.e. no newline between them, newline after).